packages/@react-stately/slider/src/useSliderState.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {clamp} from '@react-aria/utils';
+import {SliderProps} from '@react-types/slider';
+import {useControlledState} from '@react-stately/utils';
+import {useNumberFormatter} from '@react-aria/i18n';
+import {useRef, useState} from 'react';
+
+export interface SliderState {
+  // Values managed by the slider
+  readonly values: number[],
+  setValue: (index: number, value: number) => void,
+
+  // Whether a specific index is being dragged
+  isDragging: (index: number) => boolean,
+  setDragging: (index: number, dragging: boolean) => void,
+
+  // Currently-focused index
+  readonly focusedIndex: number | undefined,
+  setFocusedIndex: (index: number | undefined) => void,
+
+  // Returns the value offset as a percentage from 0 to 1.
+  getOffsetPercentForIndex: (index: number) => number,
+  getOffsetPercentForValue: (value: number) => number,
```

Comment:
These methods have potentially confusing names. Maybe combine them into a single one like `getThumbPercent(index: number, origin: number = minValue) => number`? Then you'd pass in an index of the thumb, and an optional origin to get the percentage from. What do you think?

Reply:
I did want something more general that allows me to map any value to a percent though (not just the origin, which is just one example of this use case).  For example, maybe I'd like to render tick marks for different values on the track, and would need to know where to place them, etc.

That said, we don't have to hang this off of `SliderState` (though it is convenient).  We could instead have some top-level function like this:

```tsx
// maybe part of useSlider.ts?
export function getOffsetPercentForValue(props: BaseSliderProps, value: number): number {

}
```

There's also `SliderState.getValueLabelForValue()`, which, for the same reason, would be nice to have for rendering tick marks on the track.  This will have to be off of `SliderState` instead of a top-level function, though, since it relies on the `useNumberFormatter` hook.  I think if this one is on the `SliderState`, then probably makes sense to keep the family of similar functions on `SliderState` too.

Definitely open to better names though :sweat_smile:  Maybe we could do, `getThumbPercent(index: number)` vs `getValuePercent(value: number)`?

Reply:
Makes sense. Those names sound good to me. 👍 

> There's also SliderState.getValueLabelForValue

 Maybe `getFormattedValue(value: number)`?